packages/components/src/Dialog/Layout/DialogContent.tsx

```diff
@@ -46,79 +46,49 @@ export interface DialogContentProps
    * touch the container edges.
    */
   innerProps?: SpaceProps
-
-  borderBottom?: boolean
-}
-
-interface ContentState {
-  overflow: boolean
 }
 
-interface InternalContentProps extends DialogContentProps {
+interface DialogContentLayout extends DialogContentProps {
   renderedHeight: string
 }
 
-class Internal extends Component<InternalContentProps, ContentState> {
-  private ref: RefObject<HTMLDivElement>
-
-  constructor(props: InternalContentProps) {
-    super(props)
-    this.state = { overflow: false }
-    this.ref = createRef()
-  }
-
-  public hasOverflow(e: HTMLDivElement) {
-    return e.offsetHeight < e.scrollHeight
-  }
-
-  public componentDidUpdate(prevProps: InternalContentProps) {
-    if (prevProps.renderedHeight !== this.props.renderedHeight) {
-      this.ref.current &&
-        this.setState({ overflow: this.hasOverflow(this.ref.current) })
-    }
-  }
-
-  public render() {
-    const {
-      borderBottom,
-      children,
-      className,
-      innerProps,
-      ...props
-    } = this.props
-
-    if (innerProps && innerProps.p && !innerProps.px) {
-      innerProps.px = innerProps.p
+const DialogContentLayout: FC<DialogContentLayout> = ({
+  children,
+  className,
+  innerProps,
+  renderedHeight,
+  ...props
+}) => {
+  const internalRef = useRef<HTMLDivElement>(null)
+  const [overflow, setOverflow] = useState(false)
+
+  useEffect(() => {
+    const container = internalRef.current
+
+    if (container) {
+      setOverflow(container.offsetHeight < container.scrollHeight)
     }
+  }, [renderedHeight])
 
-    return (
-      <Outer
-        borderBottom={borderBottom}
-        className={`${className} ${this.state.overflow && 'overflow'}`}
-        ref={this.ref}
-        {...omit(props, ['renderedHeight'])}
-      >
-        <Inner {...innerProps}>{children}</Inner>
-      </Outer>
-    )
+  if (innerProps && innerProps.p && !innerProps.px) {
+    innerProps.px = innerProps.p
   }
-}
 
-export const DialogContent = (props: DialogContentProps) => {
   return (
-    <ReactResizeDetector handleHeight>
-      {(height: string) => <Internal renderedHeight={height} {...props} />}
-    </ReactResizeDetector>
+    <div
+      className={`${className} ${overflow && 'overflow'}`}
```

Comment:
Now that the snapshot below has improved by ditching the `undefined` in the `className`, you could make it even better by getting rid of the `false` as well. ;)
```suggestion
      className={`${className} ${overflow ? 'overflow' : ''}`}
```